examples/plugin/cmd/main.go

```diff
@@ -0,0 +1,80 @@
+package main
+
+import (
+	"fmt"
+
+	argocd "github.com/argoproj-labs/argocd-ephemeral-access/api/argoproj/v1alpha1"
+	api "github.com/argoproj-labs/argocd-ephemeral-access/api/ephemeral-access/v1alpha1"
+	"github.com/hashicorp/go-hclog"
+
+	"github.com/argoproj-labs/argocd-ephemeral-access/pkg/log"
+	"github.com/argoproj-labs/argocd-ephemeral-access/pkg/plugin"
+	goPlugin "github.com/hashicorp/go-plugin"
+)
+
+// SomePlugin this is the struct that implements the plugin.AccessRequester
+// interface. It is preferable (but not required) to rename this struct to
+// something related to what your plugin will be doing. Example: JiraPlugin
+type SomePlugin struct {
+	Logger hclog.Logger
+}
+
+// This function will be called once when the EphemeralAccess controller is
+// initialized. It can be used to instantiate clients to other services needed by
+// this plugin for example. Those instances can then be assigned in this p for
+// later use. Just return nil of this is not required for your plugin.
+func (p *SomePlugin) Init() error {
+	p.Logger.Info("This is a call to the Init method")
+	return nil
+}
+
+// GrantAccess is the method that will be called by the EphemeralAccess controller
+// when an AccessRequest is created. The EphemeralAccess controller will only proceed
+// granting the access if the returned GrantResponse.Status is plugin.GrantStatusGranted.
+// Returning a nil GrantResponse will cause an error in the EphemeralAccess controller
+// and no access will be granted.
+// This function can be used to addresss different use-cases.
+// A few examples are:
+// - verify if the given app has an associated Change Request in approved state
+// - access internal service for last mile user access validation
+func (p *SomePlugin) GrantAccess(ar *api.AccessRequest, app *argocd.Application) (*plugin.GrantResponse, error) {
```

Comment:
I think instead of GrantAcess/RevokeAcess,  example should have `GetAcess` which can be Granted/Denied.

Reply:
`GrantAccess` is the method defined in the `AccessRequester` interface which is invoked by the EphemeralAccess controller when an `AccessRequest` is being granted. The `GrantAccess` can return `Granted/Denied` which will impact the controller differently. I updated the godoc in the example better explaining the different statuses.